libraries/botbuilder-dialogs-adaptive-testing/tests/functions.test.js

```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+const { ComponentRegistration } = require('botbuilder-core');
+const { AdaptiveComponentRegistration } = require('botbuilder-dialogs-adaptive');
+const { ResourceExplorer } = require('botbuilder-dialogs-declarative');
+const { AdaptiveTestComponentRegistration, TestUtils } = require('../lib');
+
+describe('ActionScopeTests', function () {
+    this.timeout(5000);
+
+    ComponentRegistration.add(new AdaptiveComponentRegistration());
+    ComponentRegistration.add(new AdaptiveTestComponentRegistration());
+
+    const resourceExplorer = new ResourceExplorer().addFolder(
+        path.join(__dirname, 'resources/FunctionsTests'),
+        true,
+        false
+    );
+
+    it('HasPendingActions', async () => {
+        await TestUtils.runTestScript(resourceExplorer, 'hasPendingActions');
```

Comment:
We should really consider adding a mocha specific helper like:

```typescript
export function runTestScriptMocha(scriptName: string, resourceExplorer: ResourceExplorer): void {
  it(scriptName, async () => {
    await TestUtils.runTestScript(scriptName, resourceExplorer);
  });
}
```

Not a change to handle in this PR, but something to think about.

Reply:
We can format all test cases inside `botbuilder-dialogs-adaptive-testing` using test script names as test names in another PR later.